File: lib/models/blaze_entity_lite.dart
import 'package:fileexplorer/enums/file_categories.dart';
import 'package:fileexplorer/enums/file_entity_type.dart';
import 'package:fileexplorer/enums/select_type.dart';
import 'package:flutter/material.dart';
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'blaze_entity_lite.freezed.dart';
part 'blaze_entity_lite.g.dart';

@freezed
abstract class BlazeEntityLite with _$BlazeEntityLite {
  const factory BlazeEntityLite({
    String fileEntityType,
    int filesCount,
    String group,
    bool isSymlink,
    String owner,
    String path,
    String permissions,
    int rTriplet,
    int sTriplet,
    int size,
    String symlinkPath,
    int timeStamp,
    int wTriplet,
    int xTriplet,
  }) = _BlazeEntityLite;


  factory BlazeEntityLite.fromJson(Map<String, dynamic> json) => _$BlazeEntityLiteFromJson(json);
}

File: android/app/src/main/kotlin/com/catalyst06/fileexplorer/models/BlazeEntityLite.kt
package com.catalyst06.fileexplorer.models

import com.catalyst06.fileexplorer.enums.FileEntityType

data class BlazeEntityLite(
        val fileEntityType: FileEntityType,
        val path: String,
        val size: Long = 0,
        val timeStamp: Long = 0,
        val filesCount: Int = 0,
        val sTriplet: Int = 0,
        val rTriplet: Int = 0,
        val wTriplet: Int = 0,
        val xTriplet: Int = 0,
        val permissions: String = "drx-rw-rw",
        val owner: String = "",
        val group: String = "",
        val isSymlink: Boolean=false,
        val symlinkPath: String=""
)